### PR TITLE
add long term archive

### LINKED ISSUE #89: Asymmetrische Verschlüsselung in der Archiv-Datenbank
Ursprünglich geplant für Sprint 19, 
Stand 14.05. nicht begonnen
https://miro.com/app/board/o9J_lOjvz8M=/?moveToWidget=3074457357188377022&cot=14

### LINKED ISSUE #95: Archivdatenzugang
Anforderungen dringend zu klären
@jensmuehlner hatte eMail-Verkehr zum Thema

### PR DESCRIPTION
**Pull request by epicsoft-llc**

### CODE REVIEW

File: src/main/resources/application-cloud.yml [hunk 1]
```diff
@@ -122,4 +129,20 @@ quicktest:
...
-
+  archive:
+    moveToArchiveJob:
+      cron: 0 * * * * *
+      locklimit: 1800000
+      #### 1209600 = 2 weeks
+      older-than-in-seconds: 1209600
+    hash:
+      algorithm: SHA-256
+      pepper: #### TODO set value
```
Comment (mschulte-tsi) [inline src/main/resources/application-cloud.yml:140]: set the value to ${QT_ARCHIVE_PEPPER}

File: src/main/resources/application-cloud.yml [hunk 2]
```diff
@@ -122,4 +129,20 @@ quicktest:
...
+    crypt:
+      secretLength: 50
+      secretLetters: true
+      secretNumbers: true
+      defaultRsa: RSA/ECB/OAEPwithSHA-512andMGF1Padding
+      defaultAes: AES/GCM/NoPadding
+      AesGcmNoPadding256:
+        iterations: #### TODO set value
```
Comment (mschulte-tsi) [inline src/main/resources/application-cloud.yml:148]: set the value to e.g. ${QT_ARCHIVE_ITERATIONS}

**Review by mschulte-tsi — CHANGES_REQUESTED** [review]

File: src/main/java/app/coronawarn/quicktest/config/ArchiveEntityManagerConfig.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    private HibernateJpaVendorAdapter vendorAdaptor() {
+        log.debug("Creating Archive HibernateJpaVendorAdapter");
+        final HibernateJpaVendorAdapter vendorAdapter = new HibernateJpaVendorAdapter();
+        vendorAdapter.setShowSql(true);
+        return vendorAdapter;
+    }
+
+    private Properties jpaHibernateProperties() {
```
[the remark this answers is not in the document]
Comment (epicsoft-de) [reply src/main/java/app/coronawarn/quicktest/config/ArchiveEntityManagerConfig.java]: changed

**Review by mschulte-tsi — APPROVED** [review]

File: src/main/java/app/coronawarn/quicktest/repository/QuickTestArchiveRepository.java [hunk 4]
```diff
@@ -45,4 +48,6 @@ List<QuickTestArchiveView> findAllByTenantIdAndPocIdAndTestResultInAndUpdatedAtB
             LocalDateTime dateTo
     );
 
+    Stream<QuickTestArchive> findAllByUpdatedAtBefore(
+            @Param("updatedAt") LocalDateTime updatedAtBefore, Pageable pageable);
```
[the remark this answers is not in the document]
Comment (bergmann-dierk) [reply src/main/java/app/coronawarn/quicktest/repository/QuickTestArchiveRepository.java]: Totally right, removed and renamed

File: src/main/java/app/coronawarn/quicktest/service/ArchiveService.java [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+    private final CryptionService cryptionService;
+
+    @Transactional
+    @Scheduled(cron = "${archive.moveToArchiveJob.cron}")
+    @SchedulerLock(name = "MoveToArchiveJob", lockAtLeastFor = "PT0S", 
+        lockAtMostFor = "${archive.moveToArchiveJob.locklimit}")
+    public void moveToArchiveJob() {
+        log.debug("Starting Job: moveToArchiveJob");
+        this.moveToArchive();
```
[the remark this answers is not in the document]
Comment (bergmann-dierk) [reply src/main/java/app/coronawarn/quicktest/service/ArchiveService.java]: removed

File: src/main/java/app/coronawarn/quicktest/service/HsmKeyProvider.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+@Profile({ "hsm" })
+@Service
+@RequiredArgsConstructor
+public class HsmKeyProvider implements KeyProvider {
+
+    private final ArchiveProperties properties;
+
+    @Override
+    public PublicKey getPublicKey() {
+        // TODO impl
```
[the remark this answers is not in the document]
Comment (bergmann-dierk) [reply src/main/java/app/coronawarn/quicktest/service/HsmKeyProvider.java]: replaced by vault transit engine. keeping the file without todo

**Review by f11h — APPROVED** [review]

**Review by f11h — CHANGES_REQUESTED** [review]
tbd

**Review by f11h — APPROVED** [review]

**Pull request merged**